File: src/main/java/model/Agency.java
package model;

import jakarta.persistence.*;

@Entity
@Table(name = "agencies")
public class Agency {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String username;

    @Column(nullable = false)
    private String password;

    protected Agency() {}

    public Agency(int id, String name, String username, String password) {
        this.id = id;
        this.name = name;
        this.username = username;
        this.password = password;
    }

    public int getId() { return id; }
    public String getName() { return name; }
    public String getUsername() { return username; }
    public String getPassword() { return password; }

    @Override
    public String toString() {
        return "Agency{id=" + id + ", name='" + name + '\'' + '}';
    }
}

File: TurismUI/TurismClient/Models/Agency.cs
namespace TurismClient.Models;

public sealed class Agency
{
    public int    Id       { get; }
    public string Name     { get; }
    public string Username { get; }

    public Agency(int id, string name, string username)
    {
        Id       = id;
        Name     = name;
        Username = username;
    }

    public override string ToString() => $"Agency {{ Id={Id}, Name={Name} }}";
}
